components/brave_wayback_machine/brave_wayback_machine_tab_helper.cc

```diff
@@ -103,4 +168,11 @@ bool BraveWaybackMachineTabHelper::ShouldAttachWaybackMachineInfoBar(
   return responses.find(response_code) != responses.end();
 }
 
+void BraveWaybackMachineTabHelper::CloseWindowIfPossible() {
+  if (auto* widget = views::Widget::GetWidgetForNativeWindow(active_window_)) {
+    widget->CloseWithReason(views::Widget::ClosedReason::kUnspecified);
+    active_window_ = nullptr;
+  }
+}
```

Comment:
I'm not 100% sure, it feels like it's a layering violation. I think most `//components` doesn't have views specific code and delegate it or notify events to the `//chrome` layer. What do you think?

Reply:
I realized that we don't need to handle window closing here. all deleted.
Instead, `WaybackMachineStateManager` tries to close bubble when active tab changes.
Regarding to layering, I searched `ui/views` in `components` and it seems `view` is widely used in `components`.
Maybe upstream allows to use it from components layer.